doc/nrf/protocols/matter/end_product/security.rst

```diff
@@ -1,16 +1,121 @@
+You can find an overview of the cryptography layer configuration supported for each |NCS| Matter-enabled platform in the :ref:`matter_platforms_security_support` section.
+
+Trusted storage
+***************
+
+:ref:`trusted_storage_in_ncs` is a security mechanism designed to securely store and manage sensitive data, such as cryptographic keys, device credentials, and configuration data.
+The mechanism is essential for IoT devices, as it allows the implementation of secure communication between devices.
+
+The Trusted Storage can utilize one of the following backends to implement PSA Certified Secure Storage API:
+
+* TF-M Platform Root of Trust (PRoT) - Only if the `ARM TrustZone`_ technology and hardware-accelerated firmware isolation is supported by the platform in use.
+* The :ref:`trusted_storage_readme` |NCS| software library
+
+Currently all :ref:`matter_samples` in the |NCS| use the trusted storage library as a Trusted Storage backend for all supported platforms.
+
+The trusted storage library provides confidentiality, integrity, and authenticity for the assets it operates on.
+It handles sensitive data in two contexts:
+
+* Volatile - Before data is forwarded to the non-volatile storage and after it is retrieved from the non-volatile storage.
+* Non-volatile - While data is being written to the non-volatile memory.
+
+In the case of the volatile context, the trusted storage library leverages the Authenticated Encryption with Associated Data (AEAD) encryption backend (:kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD`).
+AEAD can be configured with the set of Kconfig options described in the library's :ref:`trusted_storage_configuration` section.
+
+An important setting, that depends on the hardware platform in use, is the way of generating the AEAD keys.
+The recommended and the most secure option is to use :ref:`lib_hw_unique_key` (HUK) library.
+HUK support is automatically enabled with the :kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK` Kconfig option for compatible configurations.
+
+Currently, the :ref:`lib_hw_unique_key` library is not supported for nRF52840 Matter-enabled devices.
+Instead, in this case, AEAD keys are derived with a SHA-256 hash (:kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_HASH_UID`).
+This approach is less secure than using the library for key derivation as it will only provide integrity of sensitive material.
+It is also possible to implement a custom AEAD keys generation method when the :kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_CUSTOM` Kconfig option is selected.
+
+.. note::
+   Using HUK library in the case of nRF52840 is possible, yet it requires employing b0 bootloader that would generate the AEAD key at first boot and store it in the dedicated HUK partition that can be accessed only by the CryptoCell peripheral.
```

Comment:
```suggestion
   Using the HUK library with nRF52840 SoC is possible, but it requires employing a ``b0`` bootloader that would generate the AEAD key at first boot and store it in the dedicated HUK partition that can be accessed only by the CryptoCell peripheral.
```
Could also use :ref:`bootloader` to link to nRF Secure Immutable Bootloader if that is intended instead of ``b0``.